### DISCUSSION TITLE
A plug and play XIAO spring connector & expansion

### DISCUSSION THREAD

**Discussion by kevin-seeed** [Seeed-Studio/OSHW-XIAO-Series#6] [In Development]
If there were a connector that allowed you to use XIAO's IO pins without the need for soldering header pins, and made it easy for you to swap different XIAO modules in your designs, wouldn't that be cool? The video below presents a design for such a connector. If you're interested, please leave a comment.

https://github.com/Seeed-Studio/XIAO_Series/assets/170942971/657f9680-1620-4554-8e7d-7b0576c3edb3

**Comment by PjGLasso** [comment]
https://github.com/Seeed-Studio/XIAO_Series/assets/144751661/12cc417b-2434-4853-83da-74966bf05a72

**Reply by PjGLasso** [reply]
Print , add pins , instant NO-Solder  TEST jig!

**Reply by SEFitzstephe** [reply]
why not just sell this stl file?

**Comment by PjGLasso** [comment]
https://github.com/Seeed-Studio/XIAO_Series/assets/144751661/6c93e3b0-d82d-405a-885b-4d5b93c60912

**Comment by PjGLasso** [comment]
But wait there's more, A Xiao in a Box with the socket & battery..

https://github.com/Seeed-Studio/XIAO_Series/assets/144751661/6afc24d6-0400-4f5a-bc10-0618e8d6049d

**Reply by SEFitzstephe** [reply]
why not just sell the stl file?

**Comment by PjGLasso** [comment]
https://github.com/Seeed-Studio/XIAO_Series/assets/144751661/eab9dc98-755e-4597-beb7-dabe9f06c454

No soldering...LOL I don't think someone was prepared for the soldering on the Live stream...

**Comment by PjGLasso** [comment]
https://vimeo.com/948133413/8c4eceff94?share=copy

**Comment by Chunchun-tian** [comment]
Any suggestions on the new design of the snap connector expansion board?
![QQ_1732703319136](https://github.com/user-attachments/assets/09bb432d-340e-4ade-ab17-817a6ebe55c1)
![QQ_1732703303883](https://github.com/user-attachments/assets/e904be88-5792-42a5-aced-d6909b38fb47)

**Reply by meilily** [reply]
We also need suggestions on the product name:
Dock Expansion for XIAO?
Clip Expansion for XIAO?
XIAO Hub?
Snap Dock for XIAO?

**Reply by PjGLasso** [reply]
Man That looks Awesome...How about two Battery PINS Top Side? and a Power LED ? Some holes for Feet or make acrylic bottom like Dev board 1.
Name it ..

 **### Xiao Quick DEV**, featuring the "No Solder" Snap in Socket for Xiao... 

Tiny, Powerful, and a **SNAP** to develop with. Get yours today Free with the purchase of the complete Xiao Family Pack* or separately for the Low Xmas Price of 29.99

- Power Switch and Power LED
- Snap-in Socket for All Xiao Family of MCU's
- 4 Grove Industry Standard connectors for IO
- New "Quick" connectors for maximum compatability 
- Lipo battery power or USB-C powered.
- 2 momentary Buttons 
- EXTRA RGBW LED (for the Xiao's that lack the User LED) *wish_list
- 2 pins for battery status. (parallel with Bat connector) *wish list
- PLUG on OLED display **HAT** board Available (2Q-2025) * wish list  :-p 👍 

GO! 

HTH
PJ :v:

**Reply by PjGLasso** [reply]
" Xiao Family Pack... One of every Xiao in a nice BOX! 👍

**Reply by Lepoulpevert** [reply]
@Chunchun-tian Is this nice board already available on the market?

**Reply by SEFitzstephe** [reply]
this i would buy. when is it going to be sold?

**Comment by Chunchun-tian** [comment]
Great suggestions, I'll seriously consider these caveats. On a side note, this current version comes with on-board RGB, which is presented in an interesting way that I can show you when I get the real thing

**Comment by Chunchun-tian** [comment]
The new prototype has arrived. Any suggestions?
![b6f585410c59848c51339a5f3fbd4318_origin](https://github.com/user-attachments/assets/e43d9bb5-8812-4813-a5ab-ca10fbb5cd77)
![5c5b4a45f1e452a9c529beb92712ea4a_origin](https://github.com/user-attachments/assets/fed0057c-67ff-47c6-b3a1-5b9a780bcc28)
![e9adee1a8a76ec978e9a8e9e8ee2b11d_origin](https://github.com/user-attachments/assets/d382e151-4f57-4063-a3d4-36fbfb5d5dad)

https://github.com/user-attachments/assets/7d1f1e4a-f6c8-4bff-876f-1b7a58a6d22c

**Reply by PjGLasso** [reply]
Yes,I have a suggestion.. Send one to me IMMEDIATELY for Testing... :-)

**Reply by PjGLasso** [reply]
Man you weren't kidding, You have to see it to be impressed... WOW. words don't do it justice. what is it? 
I like the "**CLIP BOARD**" nice word work there... you guys do get it... :+1:

**Reply by Chunchun-tian** [reply]
LOL, give us some time to adjust, and send you alpha samples later for you to play with.

**Reply by iarakistain** [reply]
Is it open source?

**Comment by rei-vilo** [comment]
Answering to the kind invitation

> Some highlight features:
> 1⃣️ Spring connector for taping on your XIAO with only one clip, no solder at all

Great!

Are the Xiao Pro boards with 9 extra pins supported as well?

> 2⃣️ Power Switch and Power LED

Not a big fan of LED as it draws current.

> 3⃣️ 2 Push Buttons Reserved for you to define

Let's keep connecting the buttons an option with jumpers.

> 4⃣️ Power over Lipo battery or USB-C 

Good.

> 5⃣️ 2.54mm pin headers for maximum compatibility

Are the Xiao Pro boards with 9 extra pins supported as well?

> 6⃣️ 4 Grove Industry Standard Connectors for expanding to over 300 Grove modules
> 7⃣️ 2 [SparkFun Electronics](https://www.linkedin.com/company/sparkfun-electronics/) Qwiic Connectors reserved

Also known as [Stemma and Stemma QT](https://learn.adafruit.com/introducing-adafruit-stemma-qt/what-is-stemma)

Any connection for debugging with SWD or JTAG?

**Reply by Chunchun-tian** [reply]
Hi, currently does not support the additional 9 pins on the side of XIAO Plus, but the IO on the back is all led out.

**Comment by SleepInfinity** [comment]
Hello everyone,

I'm working on a project that uses the Xiao ESP32 S3 Plus because I need the extra GPIOs for multiple I2S components and motors—something the standard version can't handle.

I'm designing a custom PCB for my robot and want to make the Xiao board swappable, I’m looking for a way to make contact with all GPIOs, not just the through-hole ones, using side spring clips or a similar socket-style connector.

Since this topic is already being discussed here, I’d love to know if there are any updates or if anyone has suggestions on existing connectors that could work for this.

Thanks in advance!

**Comment by Lepoulpevert** [comment]
> If there were a connector that allowed you to use XIAO's IO pins without the need for soldering header pins, and made it easy for you to swap different XIAO modules in your designs, wouldn't that be cool? The video below presents a design for such a connector. If you're interested, please leave a comment.
> 2f5b070eb853ee334db09c7c48799f91.mp4

@kevin-seeed Can you share your files for this nice board?
